hack/verify-generated-files-remake.sh

```diff
@@ -21,10 +21,328 @@ set -o pipefail
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 
-make generated_files
-X=$(make generated_files DBG_CODEGEN=1)
-if [[ -n "${X}" ]]; then
-    echo "Back-to-back 'make' runs are not clean for codegen"
-    echo "${X}"
+# $1 = filename pattern as in "zz_generated.$1.go"
+function find_genfiles() {
+    find .                         \
+        \(                         \
+          -not \(                  \
+            \(                     \
+                -path ./_\* -o     \
+                -path ./.\*        \
+            \) -prune              \
+          \)                       \
+        \) -name "zz_generated.$1.go"
+}
+
+# $1 = filename pattern as in "zz_generated.$1.go"
+# $2 timestamp file
+function newer() {
+    find_genfiles "$1" | while read F; do
+        if [[ "${F}" -nt "$2" ]]; then
+            echo "${F}"
+        fi
+    done
+}
+
+# $1 = filename pattern as in "zz_generated.$1.go"
+# $2 timestamp file
+function older() {
+    find_genfiles "$1" | while read F; do
+        if [[ "$2" -nt "${F}" ]]; then
+            echo "${F}"
+        fi
+    done
+}
+
+function assert_clean() {
+    make generated_files >/dev/null
+    touch "${STAMP}"
+    make generated_files >/dev/null
+    X=($(newer deepcopy "${STAMP}"))
+    if [[ "${#X[*]}" != 0 ]]; then
+        echo "Generated files changed on back-to-back 'make' runs:"
+        echo "${X[@]:+${X[@]}}"
+        return 1
+    fi
+    true
+}
+
+STAMP=/tmp/stamp.$RANDOM
+
+#
+# Test when we touch a file in a package that needs codegen.
+#
+
+assert_clean
+
+DIR=staging/src/k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1
```

Comment:
I'm worried that this file will either get out of hand or else be neglected. You're adding over 300 lines of bash here. Is it possible to get this sort of coverage with go tests?

Related: if this test passes, do you ever expect the next to fail?

Reply:
I waffled on whether to test all generators, or just test one and assume the rest are the same.  I err on the side of caution.

Bash is not, of itself, a bad thing.  Fundamentally this is driving the calls of `make` commands.  Writing it in go is possible, but seems artificial.  Given that we have NOTHING covering this today, I'd rather go with this than nothing.  I don't have time to rewrite it right now :(

At this point we should probably wait until after we reopen for 1.10 development to merge this (if we do).

Reply:
> I waffled on whether to test all generators, or just test one and assume the rest are the same. I err on the side of caution.

My concern is that it won't stay up-to-date, so we'll be testing most generators, not all.

Reply:
most is better than none?  @pwittrock and @sttts and I were discussing maybe reducing this to a single mondo-generator, which would simplify..